src/html5/tokenizer.rs

```diff
@@ -109,6 +118,11 @@ impl<'stream> Tokenizer<'stream> {
         self.error_logger.borrow()
     }
 
+    /// Sets the tokenizer state to a new state
```

Comment:
When parsing a fragment, the tokenizer must be set to a specific state depending on the (context) node the fragment. We use a public `set_state` for this.

Reply:
I can see this changing when I start to fiddle with the state machine.  Hopefully this detail is not set in stone.

Reply:
nothing is.. but the parser NEEDS a way to set the tokenizer state. This is important, as the document fragment tokenizing does not start in the default (data) state.